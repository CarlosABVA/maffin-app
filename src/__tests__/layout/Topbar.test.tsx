import React from 'react';
import { render } from '@testing-library/react';

import type { AccountSelectorProps } from '@/components/AccountSelector';
import Topbar from '@/layout/Topbar';

jest.mock('next/navigation', () => ({
  useRouter: jest.fn(),
}));
// eslint-disable-next-line @typescript-eslint/no-var-requires
const useRouter = jest.spyOn(require('next/navigation'), 'useRouter');

jest.mock('@/components/ProfileDropdown', () => {
  function ProfileDropdown() {
    return (
      <div className="ProfileDropdown" />
    );
  }

  return ProfileDropdown;
});

jest.mock('@/components/AccountSelector', () => {
  function AccountSelector({ id, placeholder, onChange }: AccountSelectorProps) {
    // call onChange to verify it's functionality
    onChange!({ guid: '123' });
    return (
      <div className="AccountSelector">
        <span>{id}</span>
        <span>{placeholder}</span>
      </div>
    );
  }

  return AccountSelector;
});

describe('Topbar', () => {
  let mockRouterPush: jest.Mock;

  beforeEach(() => {
    mockRouterPush = jest.fn();
    useRouter.mockImplementation(() => ({
      push: mockRouterPush,
    }));
  });

  it('renders as expected', () => {
    const { container } = render(<Topbar />);

    expect(mockRouterPush).toHaveBeenCalledWith('/dashboard/accounts/123');
    expect(container).toMatchSnapshot();
  });
});
