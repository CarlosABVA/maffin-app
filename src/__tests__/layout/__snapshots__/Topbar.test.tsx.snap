// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topbar renders as expected 1`] = `
<div>
  <div
    class="grid grid-cols-12 bg-gunmetal-700 h-20 fixed z-10 top-0 px-6 left-20 right-0"
  >
    <div
      class="col-span-3"
    >
      <div
        class="AccountSelector"
      >
        <span>
          globalSearch
        </span>
        <span>
          Search (cmd + k)...
        </span>
      </div>
    </div>
    <div
      class="col-span-1 col-end-11 h-20 mb-0"
    >
      <div
        class="flex h-full justify-center items-center"
      >
        <button
          class="btn-primary"
          type="button"
        >
          ...
        </button>
      </div>
    </div>
    <div
      class="col-span-2 col-end-13 h-20"
    >
      <div
        class="ProfileDropdown"
      />
    </div>
  </div>
</div>
`;
